apps/comments-ui/src/components/content/Comment.tsx

```diff
@@ -309,8 +309,14 @@ const CommentHeader: React.FC<CommentHeaderProps> = ({comment, className = ''})
         }
 
         const element = (e.target as HTMLElement).ownerDocument.getElementById(comment.in_reply_to_id);
-        if (element) {
+        if (element && comment.parent_id) {
+            dispatchAction('setHighlightedRepliedToComment', {
+                replyToId: comment.in_reply_to_id
+            });
             element.scrollIntoView({behavior: 'smooth', block: 'center'});
+            setTimeout(() => {
+                dispatchAction('removeHighlightedRepliedToComment', null);
+            }, 2500);
```

Comment:
Feels like we're spreading knowledge of the styling behaviour around too much. Can we look at alternatives? Two ideas:
1. move the timeout and state update into `setHighlightedRepliedToComment` action so it's not necessary to remember to unset it each time the action is called
2. listen to the animation event somehow to remove the state when the animation finishes - not sure how feasible that will be based on React's rendering approach

Reply:
> For 1, I couldn't find an optimal way nor an existing pattern of modifying the state within the action, since it `returns` with the state changes.

I think adding an event listener to an animation would once again bring too much start / end knowledge in, which we want to avoid. 🤔 

Instead created a little reusable hook that will take care of the state before and after the timeout and handle all that for us.

Reply:
💡

Reply:
> For 1, I couldn't find an optimal way nor an existing pattern of modifying the state within the action, since it returns with the state changes.

That's fine? We return immediately with the new state but before the return we start a timeout that kicks off a later state change. Or am I oversimplifying? That was my intention with the likes etc, where we update state immediately after kicking off a background task that makes the API request and does a final state update.

Reply:
The problem I’ve encountered is that invoking an action within another action (as a background task) doesn’t update the state as intended since it bypasses the dispatcher. We cannot use the dispatcher (dispatchAction) within actions because it’s a hook that’s only available within React components. This seems to be a design limitation in how actions were designed.

That’s why my initial approaches involved either dispatching the setting/unsetting of the highlight within the React component or using a reusable custom hook helper to handle it. I think this is a reasonable approach since hooks are designed for managing UI updates, which is exactly what we’re doing here.

I agree with your concern about keeping the logic within the actions. However, given our constraints, handling the timeout logic in the component or a custom hook seemed to be the most practical / clearest solution for this case.

I think there's definitely room for finding a better pattern when we start working on Likes for example.

Reply:
Would it make sense to pass `dispatchAction` in as an argument to `ActionHandler` when called from `dispatchAction`? That would let us trigger actions from within actions 🤔 Although maybe allowing direct state updates inside actions is also useful for other use-cases.

Reply:
The reason I like the idea of calling actions from within actions is that it allows us to keep specific behaviours contained and re-usable, e.g. `highlightComment` and `removeHighlightComment` - although maybe not the best example usage 😅